### ISSUE TITLE
Error hook not invoked if FCMv1 endpoint returns 'NOT_FOUND'

### ISSUE THREAD

**Issue by ilmDitsch** [kayac/Gunfish#64]
Google FCMv1 endpoint returns 'NOT_FOUND' error message if token is replaced. So the error hook should be invoked to remove the old token.

**Steps to reproduce:**
1. Registered a device (and successfully sent a message via FCMv1 gateway)
2. Deleted app and reregistered device (results in new token, with the first beeing still stored)
3. Send message to the old (and now orphan) token 

**Result:**
```
time="2020-01-23T09:25:02Z" level=error msg="Unknown error message: NOT_FOUND" file=/home/circleci/project/supervisor.go line=432 res_queue_size=0 resend_cnt=0 resp_uid=07cd1248-7ee4-4d9c-9169-894cce461274 response_time=0.3715487 token="[redacted]" type=worker worker_id=0
```
(Tested with v0.4.1-rc2 Docker container: https://hub.docker.com/r/kayac/gunfish/tags?page=1&ordering=last_updated)

To me it looks like FCM is responding something like this, where the status is `NOT_FOUND` but the details errorCode is `UNREGISTERED`:
```
{
  "error": {
    "code": 404,
    "message": "Requested entity was not found.",
    "status": "NOT_FOUND",
    "details": [
      {
        "@type": "type.googleapis.com/google.firebase.fcm.v1.FcmError",
        "errorCode": "UNREGISTERED"
      }
    ]
  }
}
```

This is also discussed in similar projects facing the same issue:
- https://stackoverflow.com/questions/55689804/firebase-cloud-messaging-tokens-stop-working-404-not-found
- https://stackoverflow.com/questions/48928972/fcm-http-v1-api-returns-404-for-unregistered-token

**Comment by fujiwara** [comment]
@ilmDitsch Thank you for the reporting. Fixed by #65.  Could you try with v0.4.1-rc3 ?

**Comment by ilmDitsch** [comment]
@fujiwara I just tested my setup with the v0.4.1-rc3 docker container. It worked like a charm! The error hook was invoked as expected and I received the full error response:

```
{
    "provider": "fcmv1",
    "status": 404,
    "token": "[redacted]",
    "error": {
        "status": "NOT_FOUND",
        "message": "Requested entity was not found.",
        "details": [
            {
                "@type": "type.googleapis.com/google.firebase.fcm.v1.FcmError",
                "errorCode": "UNREGISTERED"
            }
        ]
    }
}
```
Thank you very much! It helps a lot...
